rubin_sim/maf/slicers/oneDSlicer.py

```diff
@@ -12,81 +12,94 @@
 __all__ = ['OneDSlicer']
 
 class OneDSlicer(BaseSlicer):
-    """oneD Slicer."""
+    """OneD Slicer allows the 'slicing' of data into bins in a single dimension.
+
+    Parameters
+    ----------
+    sliceColName : `str`
+        The name of the data column to base slicing on (i.e. 'airmass', etc.)
+    sliceColUnits : `str`, optional
+        Set a name for the units of the sliceCol. Used for plotting labels. Default None.
+    bins : np.ndarray, optional
+        The data will be sliced into 'bins': this can be defined as an array here. Default None.
+    binMin : `float`, optional
+    binMax : `float`, optional
+    binsize : `float`, optional
+        If bins is not defined, then binMin/binMax/binsize can be chosen to anchor the slice points.
+        Default None.
+        Priority goes: bins >> binMin/binMax/binsize >> data values (if none of the above are chosen).
+
+    The bins act like numpy histogram bins: the last bin value is the end value of the last bin.
+    All bins except for the last bin are half-open ([a, b)) while the last bin is ([a, b]).
+    """
     def __init__(self, sliceColName=None, sliceColUnits=None,
                  bins=None, binMin=None, binMax=None, binsize=None,
                  verbose=True, badval=0):
-        """
-        'sliceColName' is the name of the data column to use for slicing.
-        'sliceColUnits' lets the user set the units (for plotting purposes) of the slice column.
-        'bins' can be a numpy array with the binpoints for sliceCol or a single integer value
-        (if a single value, this will be used as the number of bins, together with data min/max or binMin/Max),
-        as in numpy's histogram function.
-        If 'binsize' is used, this will override the bins value and will be used together with the data min/max
-        or binMin/Max to set the binpoint values.
-
-        Bins work like numpy histogram bins: the last 'bin' value is end value of last bin;
-          all bins except for last bin are half-open ([a, b>), the last one is ([a, b]).
-        """
-        super(OneDSlicer, self).__init__(verbose=verbose, badval=badval)
+        super().__init__(verbose=verbose, badval=badval)
+        if sliceColName is None:
+            raise ValueError('sliceColName cannot be left None - choose a data column to group data by')
         self.sliceColName = sliceColName
         self.columnsNeeded = [sliceColName]
         self.bins = bins
-        self.binMin = binMin
-        self.binMax = binMax
-        self.binsize = binsize
-        if sliceColUnits is None:
-            co = ColInfo()
-            self.sliceColUnits = co.getUnits(self.sliceColName)
+        # Forget binmin/max/stepsize if bins was set
+        if self.bins is not None:
```

Comment:
It might be worth raising an error (or at least a warning) if a user provides both explicit bins, and also binMin etc. such that the later gets ignored.

Reply:
Good point, always good to let the user know if something potentially unexpected happened.